pallets/payment-streams/src/mock.rs

```diff
@@ -195,6 +193,23 @@ impl Convert<BlockNumberFor<Test>, Balance> for BlockNumberToBalance {
     }
 }
 
+// Mocked list of Providers that submitted proofs that can be used to test the pallet. It just returns the block number passed to it as the only submitter.
+pub struct MockSubmittingProviders;
+impl ReadProofSubmittersInterface for MockSubmittingProviders {
+    type ProviderId = <Test as frame_system::Config>::Hash;
+    type BlockNumber = BlockNumberFor<Test>;
+    type MaxProofSubmitters = ConstU32<1000>;
+    fn get_proof_submitters_for_block(
+        block_number: &Self::BlockNumber,
+    ) -> Option<BoundedBTreeSet<Self::ProviderId, Self::MaxProofSubmitters>> {
+        let mut set = BoundedBTreeSet::<Self::ProviderId, Self::MaxProofSubmitters>::new();
+        // We convert the block number + 1 to the corresponding Provider ID, to simulate that the Provider submitted a proof
+        <StorageProviders as ProvidersInterface>::get_provider_id(*block_number + 1)
+            .map(|id| set.try_insert(id));
```

Comment:
How is this supposed to simulate many providers submitting proofs?

Reply:
It does not, but I couldn't find an elegant way to do that, mostly because this registers Provider IDs which have randomness in them, which even if it's mocked it changes block by block so it's difficult to hardcode values here. For example if in a test I register Alice first and Bob second they would have different Provider IDs that if I register Bob first and Alice second